messgen/go_generator.py

```diff
@@ -103,20 +118,109 @@ def parse_string(var, t_info):
 
 messgen_types_go = {
     "char": {"fmt": fmt_int8, "parse": parse_int8, "element_type": "byte"},
-    "int8": {"fmt": fmt_int8, "parse": parse_int8},
-    "uint8": {"fmt": fmt_int8, "parse": parse_int8},
-    "int16": {"fmt": fmt_int, "parse": parse_int, "imports": ["encoding/binary"]},
-    "uint16": {"fmt": fmt_int, "parse": parse_int, "imports": ["encoding/binary"]},
-    "int32": {"fmt": fmt_int, "parse": parse_int, "imports": ["encoding/binary"]},
-    "uint32": {"fmt": fmt_int, "parse": parse_int, "imports": ["encoding/binary"]},
-    "int64": {"fmt": fmt_int, "parse": parse_int, "imports": ["encoding/binary"]},
-    "uint64": {"fmt": fmt_int, "parse": parse_int, "imports": ["encoding/binary"]},
+    "int8": {"fmt": fmt_int8, "parse": parse_int8, "stringer": Template('strconv.FormatInt(int64($varName), 10)')},
+    "uint8": {"fmt": fmt_int8, "parse": parse_int8, "stringer": Template('strconv.FormatUint(uint64($varName), 10)')},
+    "int16": {"fmt": fmt_int, "parse": parse_int, "imports": ["encoding/binary"], "stringer": Template('strconv.FormatInt(int64($varName), 10)')},
+    "uint16": {"fmt": fmt_int, "parse": parse_int, "imports": ["encoding/binary"], "stringer": Template('strconv.FormatUint(uint64($varName), 10)')},
+    "int32": {"fmt": fmt_int, "parse": parse_int, "imports": ["encoding/binary"], "stringer": Template('strconv.FormatInt(int64($varName), 10)')},
+    "uint32": {"fmt": fmt_int, "parse": parse_int, "imports": ["encoding/binary"], "stringer": Template('strconv.FormatUint(uint64($varName), 10)')},
+    "int64": {"fmt": fmt_int, "parse": parse_int, "imports": ["encoding/binary"], "stringer": Template('strconv.FormatInt($varName, 10)')},
+    "uint64": {"fmt": fmt_int, "parse": parse_int, "imports": ["encoding/binary"], "stringer": Template('strconv.FormatUint($varName, 10)')},
     "float32": {"fmt": fmt_float, "parse": parse_float, "imports": ["math", "encoding/binary"]},
     "float64": {"fmt": fmt_float, "parse": parse_float, "imports": ["math", "encoding/binary"]},
     "string": {"fmt": fmt_string, "parse": parse_string, "imports": ["$MESSGEN_MODULE_PATH/messgen"]},
 }
 
 
+def to_golang_type(_type):
+    if _type in messgen_types_go and messgen_types_go[_type].get("element_type") is not None:
+        return messgen_types_go[_type].get("element_type")
+    return _type
+
+
+def prepend_name(prefix, name):
+    if name.startswith(prefix):
+        return name
+
+    return "%s%s" % (prefix, name)
+
+
+def make_const_type(enumName, basetype, fields):
+    enum = ["type %s%s %s" % (enumName, enumSuffix, basetype)]
+
+    stringer = defaultStringer
+    if basetype in messgen_types_go and messgen_types_go[basetype].get("stringer") is not None:
+        stringer = messgen_types_go[basetype].get("stringer")
+
+    enum.append(
+        "func (v %s%s) String() string { return %s }" % (
+            enumName,
+            enumSuffix,
+            stringer.substitute(varName="v", baseType=basetype),
```

Comment:
Why using template for stringer? It's a single place, where $varName replaced by v, or I've missed something?

Reply:
because named params are better for code generation rather than positional. easier to write the template, and easier to support.